### PR TITLE
fixed FakeOutputResolver to avoid renaming correctly named nodes

### PR DESCRIPTION
**Pull request by sadolini**
Root cause analysis: Model have subgraph consist of Parameter and Result. After FakeOutputResolver and ResultRename transformations Parameter and Result names change one other. FakeOutputResolver saves result name in producer, so correct name of Parameter is lost.

Solution: Add fake operation to save result name for any producer outputs number.

Ticket: 77509

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR **NA**
* [x]  Transformation preserves original framework node names 
* [x]  Transformation preserves tensor names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests **NA**
* [x]  Transformation tests **NA**
* [x]  Model Optimizer IR Reader check **NA**

Documentation:
* [x]  Supported frameworks operations list **NA**
* [x]  Guide on how to convert the **public** model **NA**
* [x]  User guide update **NA**

### CODE REVIEW

**Review by mvafin — APPROVED** [review]

**Review by mvafin — APPROVED** [review]

File: tools/mo/openvino/tools/mo/back/FakeOutputResolver.py [hunk 1]
```diff
@@ -25,28 +25,21 @@ def find_and_replace_pattern(self, graph: Graph):
...
-                    add.out_port(0).add_tensor_names(tensor_names)
-
-            else:
-                result_in_port = fake_output.out_port(0).get_destination()
-                result_in_port.disconnect()
-                fake_output.in_port(0).get_connection().set_destination(result_in_port)
-                rename_nodes([(fake_output, name + '/TBD'), (producer, name)])
+            # At this stage we don't know the type of output, so we rely on MO transformation which updates the
+            # Const type for elementwise operations in case of input data types mismatch
+            add = create_op_with_const_inputs(graph, Add, {1: int64_array(0)}, {'can_be_fused': False})
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/back/FakeOutputResolver.py:30]: I understand that it is moved from the previous code. Can it have a type conflict with `int64`?

**Review by rkazants — APPROVED** [review]
Left one comment. Other looks good to me

File: tools/mo/openvino/tools/mo/back/FakeOutputResolver.py (hunk 1 above)
Comment (rkazants) [reply tools/mo/openvino/tools/mo/back/FakeOutputResolver.py:30]: I think there will appear warnings about it

Comment (sadolini) [reply tools/mo/openvino/tools/mo/back/FakeOutputResolver.py:30]: yes, we have warning about conversion int64 to float32 (for float output)

**Comment by artkuli** [comment]
Continuation in #11938

**Pull request closed without merging**